kratos/processes/apply_double_table_process.h

```diff
@@ -0,0 +1,100 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ignasi de Pouplana
+//  Collaborator:    Alejandro Cornejo
+//
+
+#if !defined(KRATOS_APPLY_DOUBLE_TABLE_PROCESS_H_INCLUDED)
+#define  KRATOS_APPLY_DOUBLE_TABLE_PROCESS_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "processes/apply_component_table_process.h"
+
+namespace Kratos
+{
+
+/// The base class for all processes in Kratos.
+/** This function applies a value according to a table defined in the mdpa
+*/
+class KRATOS_API(KRATOS_CORE) ApplyDoubleTableProcess : public ApplyComponentTableProcess
```

Comment:
Please, use templates to avoid code duplication, this is a blocker

Reply:
Now it's only one templated class